f5/bigip/resource.py

```diff
@@ -1035,3 +1043,84 @@ def load(self, **kwargs):
         newinst = self._stamp_out_core()
         newinst._refresh(**kwargs)
         return newinst
+
+
+class Stats(PathElement):
+    """This class allows attaching 'stats' object as an
+
+        attribute to a resource.
+
+        Stats are available on some objects, and require
+        the target object to be loaded or created first.
+
+        It also allows the caller to retrieve the converted json
+        response in unprocessed state.
+    """
+    def __init__(self, container):
+        super(Stats, self).__init__(container)
+        self._update_stats(self._get_stats_raw())
+
+    @property
+    def stats_raw(self):
+        """Provides JSON object converted to a python dictionary"""
+        return self._get_stats_raw()
+
+    def _key_dot_replace(self, rdict):
+        """Replace fullstops in returned keynames"""
+        temp_dict = {}
+        for key, value in rdict.items():
+            if isinstance(value, dict):
+                value = self._key_dot_replace(value)
+            temp_dict[key.replace('.', '_')] = value
+        return temp_dict
+
+    def _get_nest_stats(self, rdict):
+        """Helper method to deal with nestedStats
+
+        as json format changed in v12.x
+        """
+        for x in rdict:
+            check = urlparse.urlparse(x)
+            if check.scheme:
+                nested_dict = rdict[x]['nestedStats']
+                tmp_dict = nested_dict['entries']
+                return self._key_dot_replace(tmp_dict)
+
+        return self._key_dot_replace(rdict)
+
+    def _get_stats_raw(self):
+        """Displays JSON object transformed into python dictionary"""
+        read_session = self._meta_data['bigip']._meta_data['icr_session']
+        base_uri = self._meta_data['container']._meta_data['uri'] + 'stats/'
+        response = read_session.get(base_uri)
+        rdict = response.json()
+        return rdict
+
+    def _update_stats(self, rdict):
+        """Attaches stat attribute to stats object"""
+        if 'entries' not in rdict:
```

Comment:
Maybe this kind of logic needs to move into a new class in a different module.

Do we need a `response_handlers.py`?